frontend/src/components/Answer/AnswerParser.tsx

```diff
@@ -19,7 +19,7 @@ export function parseAnswer(answer: AskResponse): ParsedAnswer {
         // Replacing the links/citations with number
         let citationIndex = link.slice(lengthDocN, link.length - 1);
         let citation = cloneDeep(answer.citations[Number(citationIndex) - 1]) as Citation;
```

Comment:
What happens if citations is null or empty?

Reply:
It looks like `citations` should at least be an empty array. When it is an empty array querying it results in `citation` being undefined.